packages/abi-typegen/src/templates/predicate/factory.hbs

```diff
@@ -57,11 +57,17 @@ export class {{capitalizedName}}__factory {
   static readonly abi = _abi
   static readonly bin = _bin;
 
-  static createInstance(provider: Provider, configurables?: {{capitalizedName}}Configurables) {
+  static createInstance(provider: Provider, predicateData?: InputValue[], configurables?: {{capitalizedName}}Configurables) {
```

Comment:
I just noticed our typegen template doesn't consider the generic type, but should it? 🤔

Reply:
I think if you pass in `predicateData`, the class will automatically infer the type no? 🤔

Reply:
> I think if you pass in predicateData, the class will automatically infer the type no? 🤔

I don't think so. The way it is currently, the returned predicate type would be `Predicate<InputValue[]>`.
I believe that it should be something like this for type inference to work properly:

```hbs
static createInstance<Data extends InputValue[]>(provider: Provider, predicateData?: Data, configurables?: {{capitalizedName}}Configurables) {
    const { abi, bin } = {{capitalizedName}}__factory

    return new Predicate<Data>({
      bytecode: bin,
      abi,
      provider,
      inputData: predicateData,
      configurableConstants: configurables,
    })
}
```

Reply:
Created an issue for this: https://github.com/FuelLabs/fuels-ts/issues/1850